docs/semantic_web_guide.md

```diff
@@ -0,0 +1,715 @@
+    
+    ;; Apply reasoning
+    (def reasoned @(fluree/reason (fluree/db ledger) :owl2rl))
+    
+    ;; Query with SPARQL
+    @(fluree/query reasoned
+                   "PREFIX ex: <http://example.org/>
+                    PREFIX schema: <http://schema.org/>
+                    
+                    SELECT ?article ?type ?headline ?author
+                    WHERE {
+                      ?article a ?type ;
+                               schema:headline ?headline ;
+                               schema:author ?author .
+                      ?type rdfs:subClassOf* ex:Article .
+                    }"
+                   {"format" :sparql})))
+```
+
+### Example 2: Temporal Semantic Web Data
+
+```clojure
+;; Track changes to RDF data over time
+(defn temporal-rdf-example []
+  (let [conn @(fluree/connect-memory)
+        ledger @(fluree/create conn "temporal-rdf")]
+    
+    ;; Initial state
+    (def t1-db @(fluree/insert! ledger (fluree/db ledger)
+                                "@prefix foaf: <http://xmlns.com/foaf/0.1/> .
+                                 @prefix ex: <http://example.org/> .
+                                 
+                                 ex:project1 a ex:Project ;
+                                            ex:status \"active\" ;
+                                            ex:lead ex:alice ."
+                                {"format" "turtle"}))
+    
+    ;; Update status
+    (Thread/sleep 1000)
+    (def t2-db @(fluree/update! ledger t1-db
+                                "@prefix ex: <http://example.org/> .
+                                 
+                                 ex:project1 ex:status \"completed\" ."
+                                {"format" "turtle"}))
+    
+    ;; Query history with SPARQL
+    @(fluree/query (fluree/db ledger)
+                   "PREFIX ex: <http://example.org/>
+                    PREFIX f: <https://ns.flur.ee/ledger#>
+                    
+                    SELECT ?time ?status
+                    WHERE {
+                      GRAPH ?commit {
+                        ex:project1 ex:status ?status .
+                      }
+                      ?commit f:time ?time .
+                    }
+                    ORDER BY ?time"
+                   {"format" :sparql})))
+```
+
+## Best Practices
+
+### 1. Namespace Management
+Always define clear namespace prefixes and use consistent URIs across your data:
+
+```turtle
+@prefix : <http://example.org/vocab/> .
+@prefix data: <http://example.org/data/> .
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+```
+
+### 2. SHACL Design Patterns
+- Use SHACL for data validation
+- Define reusable shapes with `sh:node`
+- Leverage SPARQL-based constraints for complex validation rules
+
+### 3. Reasoning Performance
+- Apply reasoning selectively using graph patterns
+- Pre-compute common inferences
+- Use property paths in SPARQL instead of reasoning when possible
```

Comment:
Our SPARQL (and JLD-query) path support is limited: we support sequences of predicate path segments, inverse path segments, and we only support a single segment of one-or-more, zero-or-more, or zero-or-one paths. We do not negated paths at all.